frontend/src/hooks/queryBuilder/useTagValidation.ts

```diff
@@ -19,7 +19,9 @@ export const useTagValidation = (
 	result: string[],
 ): ITagValidation => {
 	const operatorType = useOperatorType(operator);
-	const isValidTag = useIsValidTag(operatorType, result.length);
+	const resultLength =
+		operatorType === 'SINGLE_VALUE' ? [result]?.length : result?.length;
+	const isValidTag = useIsValidTag(operatorType, resultLength);
 
 	const { isExist, isValidOperator, isMulti, isFreeText } = useMemo(() => {
 		const isExist = operatorType === QUERY_BUILDER_SEARCH_VALUES.NON;
```

Comment:
There is `checkStringEndsWithSpace`. Why is that needed? What does it decide?

Reply:
@srikanthccv - Earlier It was used for IN/NOT_IN for space around options you select. but not it's no more used. As Space for IN/NOT_IN is mandatory.
https://github.com/SigNoz/signoz/pull/2629/commits/a7ba777ca05063683fb38cc44dbbda2316e16d51